docs/run_test/azure_auth.rst

```diff
@@ -0,0 +1,151 @@
+Use Different Authentication Methods on Azure
+=============================================
+
+-  `Overview <#overview>`__
+-  `Run on Azure <#run-on-azure>`__
+-  `Available Authentication Methods <#available-authentication-methods>`__
+-  `Schema Description <#schema-description>`__
+
+Overview
+--------
+This document describes how to configure and use different authentication methods for running LISA tests on Azure. The available authentication methods include default credentials, certificates, client assertions, client secrets, workload identity, and tokens.
+You can configure the `platform` section of your YAML file to choose the desired authentication method for connecting to Azure.
+
+Run on Azure
+-------------
+To run LISA tests on Azure with different authentication methods, configure the `platform` section in your YAML file as shown below.
+
+Available Authentication Methods
+-------------------------------
+1. `Default Credentials <#default-credentials>`
+2. `Certificate Authentication <#certificate-authentication>`
+3. `Assertion Authentication <#assertion-authentication>`
+4. `Workload Identity Authentication <#workload-identity-authentication>`
+5. `Token Authentication <#token-authentication>`
+6. `Client Secret Authentication <#client-secret-authentication>`
+
+Default Credentials
+-------------------
+Default authentication uses credentials from environment variables, Azure CLI, or managed identities. This method is useful when your environment is already authenticated through Azure CLI, Azure PowerShell, or managed identities.
+
+Example:
+.. code:: yaml
+
+   platform:
+     - type: azure
+       azure:
+         credential:
+           type: default
+           client_id: <client id>  # Optional
+           tenant_id: <tenant id>  # Optional
+           allow_all_tenants: false | true  # Optional. Default is `false`.
+
+* **type**: `default` indicates default credential authentication.
+* **client_id**: (Optional) Needed when there are multiple managed identities associated with the running machine.
+* **tenant_id**: (Optional) Needed when you have multiple tenants.
+* **allow_all_tenants**: (Optional) Specifies whether to allow cross-tenant authorization. Default is `false`.
+
+**Note:** The `default` authentication type defines a base schema, which is inherited by all other authentication methods. Fields such as `allow_all_tenants` are applicable to all authentication types.
```

Comment:
default is not the base class. Default is to try all auth method as best effort. The default is not related with other auth methods. For `allow_all_tenants`, it needs to list in other classes, if a class uses the field.